tests/integration/plugins/test_jlink.py

```diff
@@ -23,6 +23,73 @@
 from craft_parts import LifecycleManager, Step, errors
 
 
+@pytest.fixture()
+def build_test_jar():
+    Path("Test.java").write_text(
+        """
+            public class Test {
+                public static void main(String[] args) {
+                    new Embedded();
+                }
+            }
+        """
+    )
+    Path("Embedded.java").write_text(
+        """
+            import javax.swing.*;
+            public class Embedded {
+                public Embedded() {
+                    new JFrame("foo").setVisible(true);
+                }
+            }
+
+        """
+    )
+    subprocess.run(
+        ["javac", "Test.java", "Embedded.java"], check=True, capture_output=True
+    )
+    subprocess.run(
+        ["jar", "cvf", "embedded.jar", "Embedded.class"],
+        check=True,
+        capture_output=True,
+    )
+    subprocess.run(
+        ["jar", "cvf", "test.jar", "Test.class", "embedded.jar"],
+        check=True,
+        capture_output=True,
+    )
+
+
+@pytest.mark.usefixtures("build_test_jar")
+def test_jlink_plugin_embedded_jar(new_dir, partitions):
+    parts_yaml = textwrap.dedent(
+        """
+        parts:
+            my-part:
+                plugin: jlink
+                source: .
+                source-branch: ubuntu-24.04
```

Comment:
This line is causing the integration test to fail, you should be able to drop it.
```suggestion
```